src/realm/object-store/sync/async_open_task.cpp

```diff
@@ -118,4 +116,53 @@ void AsyncOpenTask::unregister_download_progress_notifier(uint64_t token)
         m_session->unregister_progress_notifier(token);
 }
 
+void AsyncOpenTask::attach_to_subscription_initializer(AsyncOpenCallback&& async_open_callback,
+                                                       std::shared_ptr<_impl::RealmCoordinator> coordinator,
+                                                       bool rerun_on_launch)
+{
+    // Subscription initialization will occure in either of these 2 cases.
+    // 1. The realm file has been created (this is the first time we ever open the file)
+    // 2. we are instructed to run the subscription initializer via sync_config->rerun_init_subscription_on_open.
+    //    But we do that only if this is the first time we open realm.
+
+    if (m_db_first_open || (rerun_on_launch && m_db_first_open)) {
```

Comment:
Note: `m_db_first_open` is always false on the right-hand side of `||`.

The right-hand side expression is correct, but the left side should check if the realm was just created. I gave it a bit of thought and I think something along the lines below should work:
```
auto shared_realm = coordinator->get_realm();
auto latest_sub_set = shared_realm->get_latest_subscription_set();
if (latest_sub_set.version() == 1 || (rerun_on_launch && m_db_first_open)) {
   latest_sub_set.get_state_change_notification(sync::SubscriptionSet::State::Complete)
   ...
}

```